sklearn/datasets/kddcup99.py

```diff
@@ -0,0 +1,403 @@
+    focusing on the intrusion attack, which gives a proportion of 0.3% of
+    attack.
+    -http and smtp are two subsets of SF corresponding with third feature
+    equal to 'http' (resp. to 'smtp')
+
+    [1] : Analysis and Results of the 1999 DARPA Off-Line Intrusion Detection
+    Evaluation Richard Lippmann, Joshua W. Haines, David J. Fried, Jonathan
+    Korba, Kumar Das
+
+    [2] A Geometric Framework for Unsupervised Anomaly Detection: Detecting
+    Intrusions in Unlabeled Data (2002) by Eleazar Eskin, Andrew Arnold,
+    Michael Prerau, Leonid Portnoy, Sal Stolfo
+
+    General KDD structure :
+    ==============      =========================================
+    Samples total       4898431
+    Dimensionality      41
+    Features            discrete (int) or continuous (float)
+    Targets             str, 'normal' or name of the anomaly type
+    ==============      =========================================
+
+    SA structure :
+    ==============      =========================================
+    Samples total       976158
+    Dimensionality      41
+    Features            discrete (int) or continuous (float)
+    Targets             str, 'normal' or name of the anomaly type
+    ==============      =========================================
+
+    SF structure :
+    ==============      =========================================
+    Samples total       699691
+    Dimensionality      40
+    Features            discrete (int) or continuous (float)
+    Targets             str, 'normal' or name of the anomaly type
+    ==============      =========================================
+
+    http structure :
+    ==============      =========================================
+    Samples total       619052
+    Dimensionality      39
+    Features            discrete (int) or continuous (float)
+    Targets             str, 'normal' or name of the anomaly type
+    ==============      =========================================
+
+    smtp structure :
+    ==============      =========================================
+    Samples total       95373
+    Dimensionality      39
+    Features            discrete (int) or continuous (float)
+    Targets             str, 'normal' or name of the anomaly type
+    ==============      =========================================
+
+    Parameters
+    ----------
+    subset : None, 'SA', 'SF', 'http', 'smtp'
+        To return the corresponding classical subsets of kddcup 99
+        If None, return the entire kddcup 99 dataset.
+
+    random_state : int, RandomState instance or None, optional (default=None)
+        Random state for shuffling the dataset.
+        If int, random_state is the seed used by the random number generator;
+        If RandomState instance, random_state is the random number generator;
+        If None, the random number generator is the RandomState instance used
+        by `np.random`.
+
+    shuffle : bool, default=False
+        Whether to shuffle dataset.
+
```

Comment:
`percent10` is not documented in `load_kddcup99`
`percent10` and `subset` are not documented in `fetch_kddcup99`